### ISSUE TITLE
Textarea inside Popover loses focus on first key stroke

### ISSUE THREAD

**Issue by Maadtin** [chakra-ui/chakra-ui#2120]
# Bug report

## Describe the bug

Textarea inside Popover loses focus on first key stroke using state to control the value of the textarea

## To reproduce

1. Use Textarea component inside Popover
2. Type inside Textarea updating It's state
3. Textarea loses focus on first key stroke

## Minimal reproduction

https://codesandbox.io/s/chakra-ui-027fq?file=/src/App.js

## Expected behavior

Textarea should not lose focus when typing for the first time.

## System information

- OS: Windows
- Browser (if applies): Chrome
- Version of @chakra-ui/core: 0.8.0
- Version of Node.js: 12

**Comment by Maadtin** [comment]
Bump,  does anyone know how to fix this? :S

**Comment by candicecz** [comment]
This works for me:
```
<FocusLock>
   <TextArea>Something...</TextArea>
</FocusLock>
```
I updated your example here: https://codesandbox.io/s/chakra-ui-forked-mj91p?file=/src/App.js:753-762

**Comment by candicecz** [comment]
There's some helpful information in the docs about this here: https://chakra-ui.com/popover#trapping-focus-within-popover

**Comment by Maadtin** [comment]
It works but Why does it lose It's focus? I've noticed it happens on normal html textarea too. Also ReactFocusLock has another problem with the scrolling even thoug Im setting the  `returnFocus={{preventScroll: true}}` It still scrolls to the element... @candicecz Can you pls help?

**Comment by ljosberinn** [comment]
closing in favor of #2111

**Comment by jvnlwn** [comment]
FocusTrap didn't quite work for me, but might have been incorrect implementation on my part. When using FocusTrap, I was able to control an input value without losing focus when typing, but the value of the input would oddly be cleared out after interacting with some other part of the UI.

What I found that did provide value control (albeit "partial" control) **without** losing focus **and** prevented the mysterious behavior of the value being cleared was to store the input value in a ref and use that value as the input's `defaultValue`, thus controlling the input value on render, storing the input value for later form submission, and avoiding the need for FocusTrap.

Here's a baisc example:
```javascript
const Example = () => {
  const inputValueRef = useRef()

  return (
    <Input
      type="text"
      defaultValue={inputValueRef.current}
      onChange={e => (inputValueRef.current = e.target.value)}
    />
  )
}
```
